### ISSUE TITLE
Default range for rectbin favors rounded ticks in legend over color differentiation

### ISSUE THREAD

**Issue by garborg** [dcjones/Gadfly.jl#469]
The way you handle the range and legend is usually great, but it can leave the more distinctive colors in the outer two quartiles almost completely unused, which leaves the plot muted, and the extreme values (which are often the most interesting) difficult to identify (intermediate values are tougher than normal to identify, as well). E.g.:

``` julia
using DataFrames, Gadfly

function f(min, step)
    df = DataFrame(x = rep(1:2, 2),
                   y = rep(1:2, each=2),
                   val = min:step:(min + 3*step))

    plot(df, x="x", y="y", color="val", Geom.rectbin)
end

f(-4, 20) #=    -4 to 56
                chosen: -20:60 (max. tail: 16/80)
                lil' better: -10 10 30 50 70 (max. tail: 14/80)
                better: -10 5 20 35 50 65 (max. tail: 9/75) =#

f(199, 70) #=   -199 to 409
                chosen: 100:500 (max. tail: 99/400)
                lil' better: 150 250 350 450 (max. tail: 49/300)
                better: 175 225 275 325 375 425 (max. tail: 24/250) =#

f(199, 35) #=   199 through 304
                chosen: 150:350 (max. tail: 49/200)
                lil' better: 175 225 275 325 (max. tail: 24/150)
                better: 190 210 230 270 290 310 (max. tail: 9/120) =#
```

**Comment by dcjones** [comment]
There a few things it tries to balance when choosing ticks: number of ticks, span, "niceness" of the numbers, and the inclusion of zero.

If I add a higher weight to span, I can get ticks closer to what you'd call better. It still tends to avoid cases like `-10 10 30 50 70`, since it would rather include 0.

I should make these parameters modifiable using `Theme` at the very least, and maybe bump the default weight for span. I'll draw some plots and see how it compares on average.

**Comment by garborg** [comment]
I agree that the numbers I posted are only arguably better -- in unfortunate cases where almost everything is colored using the two middle quartiles, probably, but arguable generally.

Modifiable probably sounds good -- I hate to suggest increased complexity, but making it modifiable provides a road to forcing two plots, that would otherwise get different ticks, onto the same color scale (assuming it doesn't exist since I looked for it a while ago).
